src/hcupc_spmd.cpp

```diff
@@ -137,13 +141,7 @@ int barrier() {
  * This is adapted from:
  * Olivier, S., Prins, J.: Scalable dynamic load balancing using upc. In: ICPP. pp.123�131 (2008)
  */
-upcxx::shared_var<int> cb_cancel;
-upcxx::shared_var<int> cb_count;
-upcxx::shared_var<int> cb_done;
-upcxx::shared_lock cb_lock;
-bool workAvailInit = false;
-#define CB_LOCK		cb_lock.lock()
-#define CB_UNLOCK	cb_lock.unlock()
+static upcxx::shared_array<upcxx::atomic<unsigned int > > cb_count;
```

Comment:
I don't actually remember making this change, so this may be something older from when I was investigating the old SEGFAULTs. We may want to revert this.